File: src/python/finished/check_if_digits_are_equal_in_string_after_operations_i.py
# https://leetcode.com/problems/check-if-digits-are-equal-in-string-after-operations-i


class Solution:
    def hasSameDigits(self, s: str) -> bool:
        digits = [int(ch) for ch in s]
        bound = len(digits) - 1

        while bound >= 2:
            for i in range(bound):
                digits[i] = (digits[i] + digits[i + 1]) % 10
            bound -= 1

        return digits[0] == digits[1]

File: src/rust/finished/check_if_digits_are_equal_in_string_after_operations_i.rs
impl Solution {
    pub fn has_same_digits(s: String) -> bool {
        let mut digits: Vec<_> = s.chars().map(|ch| ch.to_digit(10).unwrap()).collect();
        let mut bound = digits.len() - 1;

        while bound >= 2 {
            for i in 0..bound {
                digits[i] = (digits[i] + digits[i + 1]).rem_euclid(10);
            }
            bound -= 1;
        }

        digits[0] == digits[1]
    }
}
